presto-spi/src/main/java/com/facebook/presto/spi/type/TypeSignature.java

```diff
@@ -217,8 +211,7 @@ else if (c == ')') {
         verify(parameters.size() == fieldNames.size() || fieldNames.isEmpty(), "Number of parameters and fieldNames for ROW type doesn't match");
         List<TypeSignatureParameter> result = new ArrayList<>();
         for (int i = 0; i < parameters.size(); i++) {
-            // Use "field" + i instead of String.format to avoid expensive string formatting
-            String fieldName = fieldNames.isEmpty() ? ("field" + i) : fieldNames.get(i);
+            Optional<String> fieldName = fieldNames.isEmpty() ? Optional.empty() : Optional.of(fieldNames.get(i));
```

Comment:
it is a pity that `empty list` means `no list`. Can we change this method to accept `Optional<List<String>> fieldNames`